site/en/docs/web-platform/bfcache-notrestoredreasons/index.md

```diff
@@ -0,0 +1,216 @@
+    console.log(`Navigation entry ${i}`);
+    let navEntry = navEntries[i];
+    console.log(navEntry.notRestoredReasons);
+  }
+}
+```
+
+The `PerformanceNavigationTiming.notRestoredReasons` property returns an object with the following structure, which represents the blocked state of the top-level frame:
+
+```js
+{
+  blocked: true,
+  children: [],
+  id: "",
+  name: "",
+  reasons: [ "Internal Error" ],
+  src: "",
+  url: "a.com"
+}
+```
+
+The properties are as follows:
+
+`blocked`
+: A boolean value specifying whether the navigated page is blocked from using the BFCache (`true`) or not (`false`).
+
+`childen`
+: An array of objects representing the blocked state of any frames embedded in the top-level frame. Each object has the same structure as the parent object — this way, any number of levels of embedded frames can be represented inside the object recursively. If the frame has no children, the array will be empty.
+
+`id`
+: A string representing the `id` attribute value of the frame (for example `<iframe id="foo" src="...">`). If the frame has no `id`, the value will be an empty string.
+
+`name`
+: A string representing the `name` attribute value of the frame (for example `<iframe name="bar" src="...">`). If the frame has no `name`, the value will be an empty string.
+
+`reasons`
+: An array of strings each representing a reason why the navigated page was blocked from using the BFCache. There are many different reasons why blocking could occur. See the [Blocking reasons](#blocking-reasons) section below for more details.
+
+`src`
+: A string representing the path to the frame's source (for example `<iframe src="b.html">`). If the frame has no `src`, the value will be an empty string.
+
+`url`
+: A string representing the URL of the navigated page.
+
+### Reporting BFCache blocking in same-origin frames
+
+When a page has same-origin frames embedded, the returned `notRestoredReasons` value will contain an object inside the `children` property representing the blocked state of each embedded frame.
+
+For example:
+
+```js
+{
+  blocked: false,
+  children: [
+    { url: "a.com", src: "b.a.com", id: "b", name: "b", blocked: false, reasons: [], children: [] },
+    { url: "a.com", src: "c.a.com", id: "c", name: "c", blocked: true, reasons: [ "BroadcastChannel" ], children: [] },
+    { url: "a.com", src: "d.a.com", id: "d", name: "d", blocked: false, reasons: [], children: [] }
+  ],
+  id: "",
+  name: "",
+  reasons: [],
+  src: "",
+  url:"a.com"
+}
+```
+
+### Reporting BFCache blocking in cross-origin frames
+
+When a page has cross-origin frames embedded, we limit the amount of information shared about them to avoid leaking cross-origin information. We only include information that the outer page already knows, and whether the cross-origin subtree blocked BFCache or not. We don't include any blocking reasons, or information about lower levels of the subtree (even if some sub-levels are same-origin).
```

Comment:
```suggestion
### Reporting bfcache blocking in cross-origin frames

When a page has cross-origin frames embedded, we limit the amount of information shared about them to avoid leaking cross-origin information. We only include information that the outer page already knows, and whether the cross-origin subtree blocked bfcache or not. We don't include any blocking reasons, or information about lower levels of the subtree (even if some sub-levels are same-origin).
```